src/mmw/apps/bigcz/clients/cuahsi.py

```diff
@@ -45,16 +77,43 @@ def find_service(services, service_code):
     return None
 
 
-def parse_records(sites, services):
+def parse_records(series, services):
+    """
+    Join series catalog records to corresponding service.
+    """
     result = []
-    for site in sites:
-        service = find_service(services, site['servCode'])
+    for record in series:
+        service = find_service(services, record['ServCode'])
         if service:
-            record = parse_record(site, service)
+            record = parse_record(record, service)
             result.append(record)
     return result
 
 
+def group_series_by_location(series):
```

Comment:
The implementation does seem faithful to the description. Perhaps because of exposure to [`_.groupBy`](https://lodash.com/docs/4.17.4#groupBy), I assumed the function did something like this:

```python
items = [
    {'id': 1, 'color': 'red'},
    {'id': 2, 'color': 'red'},
    {'id': 3, 'color': 'blue'}]

def group_list_by_color(items):
    colors = {}
    for item in items:
        color = item['color']
        if color in colors:
            colors[color].append(item)
        else:
            colors[color] = [item]

    return colors
```

So that the output is grouped by a certain key but the value is the original list item itself:

```ipython
In [3]: group_list_by_color(items)
Out[3]:
{'blue': [{'color': 'blue', 'id': 3}],
 'red': [{'color': 'red', 'id': 1}, {'color': 'red', 'id': 2}]}
```

What it's actually doing is more like extracting distinct locations from the series rather than grouping the series by location. Does that make sense?